chp07_CA/GameOfLifeWrapAround/GOL.py

```diff
@@ -0,0 +1,67 @@
+# The Nature of Code
+# Daniel Shiffman
+# http://natureofcode.com
+
+class GOL():
+
+    W = 8
+
+    def __init__(self):
+        # Initialize self.rows, self.cols and set-up a list of lists
+        self.cols = width / GOL.W
+        self.rows = height / GOL.W
+        # Game of life board
+        self.board = [[0] * self.rows for _ in range(self.cols)]
+        # Call function to fill array with random values 0 or 1
+        self.init()
+
+    def init(self):
+        for i in range(self.cols):
+            for j in range(self.rows):
+                self.board[i][j] = int(random(2))
+
+    # The process of creating the new generation
+    def generate(self):
+
+        next = [[0] * self.rows for _ in range(self.cols)]
+
+        # Loop through every spot in our 2D array and check spots neighbors
+        for x in range(self.cols):
+            for y in range(self.rows):
+                # Add up all the states in a 3x3 surrounding grid
+                neighbors = 0
+                for i in range(-1, 2):
+                    for j in range(-1, 2):
+                        nx = (x + i + self.cols) % self.cols
+                        ny = (y + j + self.rows) % self.rows
+                        neighbors += self.board[nx][ny]
+                # A little trick to subtract the current cell's state since
+                # we added it in the above loop
+                neighbors -= self.board[x][y]
+                # Rules of Life
+                # Loneliness
+                if self.board[x][y] == 1 and neighbors < 2:
+                    next[x][y] = 0
+                # Overpopulation
+                elif self.board[x][y] == 1 and neighbors > 3:
```

Comment:
but next[x][y] already is 0

Reply:
Hi jdf!
I suppose this is a difficult discussion... This is Shiffman making the Game of Life rules very explicit. Options I see:
1. We could initialize the next board with None instead of 0s;
2. We could add a comment warning this is not really necessary, as the next board is already full of 0s.
But I don't see how I could remove the explicit "die" rules without changing Shiffman's didactic intentions. Third option, comment out? 
Thanks for being such a careful a reviewer.

Reply:
I think you're right; disregard my comment.